File: problems/incomplete/1_Two_Sum/two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        /*
        
        */
        
        return null;
    }
    
    public static void main(String[] args) {
        Solution solution = new Solution();
        
        Object[][] testCases = {
            {new int[]{2,7,11,15}, 9},
            {new int[]{3,2,4}, 6},
            {new int[]{3,3}, 6}
        };
        
        for (int i = 0; i < testCases.length; i++) {
            System.out.println("___ NO." + i + " ___________________________________");
            System.out.println("Input: " + java.util.Arrays.toString((int[])testCases[i][0]) + ", " + testCases[i][1]);
            System.out.println("Output: " + java.util.Arrays.toString(solution.twoSum((int[])testCases[i][0], (int)testCases[i][1])));
            System.out.println();
        }
    }
}

File: problems/incomplete/1_Two_Sum/two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        
        '''
        
        '''



        return None



    

if __name__ == '__main__':

    cases = [
        [2,7,11,15],
		9,
		[3,2,4],
		6,
		[3,3],
		6
    ]

    #> OPTION 2 (for multiple inputs)
    s = Solution()
    for i in range(0, len(cases), 2):

        nums = cases[i+0]
        target = cases[i+1]
        print(f"___ NO.{i} ___________________________________")
        print(f"n={i} -> {s.twoSum(nums, target)}\n")



'''
(NEW) TESTCASES:
cases = [
    '[2,7,11,15]',
    '9',
    '[3,2,4]',
    '6',
    '[3,3]',
    '6',
    [1, 2, 3],
    0,
    [1, 2, 3],
    0,
    [1, 2, 3],
    0,
    [1, 2, 3],
    0,
    [1, 2, 3],
    0,
    [1, 2, 3],
    0,
    [1, 2, 3],
    0,
    [1, 2, 3],
    0,
    [1, 2, 3],
    0,
    [1, 2, 3],
    0,
]

FOR LEETCODE:
'[2,7,11,15]'
'9'
'[3,2,4]'
'6'
'[3,3]'
'6'
[1, 2, 3]
0
[1, 2, 3]
0
[1, 2, 3]
0
[1, 2, 3]
0
[1, 2, 3]
0
[1, 2, 3]
0
[1, 2, 3]
0
[1, 2, 3]
0
[1, 2, 3]
0
[1, 2, 3]
0
'''


"""
__ GITHUB PUSH COMMENT _________________________
Finish 1. Two Sum + move to completed
contains: description, solution.
difficulty: Easy
topics: Array, Hash Table
"""